frontend/ui/widget/spinwidget.lua

```diff
@@ -24,7 +24,7 @@ local SpinWidget = InputContainer:new{
     title_text = "",
     title_face = Font:getFace("x_smalltfont"),
     info_text = nil,
-    width = nil,
+    width = nil, -- width, or when between 0 and 1, factor to the smallest of screen width and height
```

Comment:
So, to avoid the deceptiveness and have it both options explicite, that would become:
```lua
    width = nil,
    width_factor = nil, -- number between 0 and 1, factor to the smallest of screen width and height
```